app/src/androidTest/java/com/github/sdp_begreen/begreen/utils/PermissionsTest.kt

```diff
@@ -0,0 +1,42 @@
+package com.github.sdp_begreen.begreen.utils
+
+import android.Manifest
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.rule.GrantPermissionRule
+import com.github.sdp_begreen.begreen.activities.SignInActivity
+import com.github.sdp_begreen.begreen.utils.Permissions.hasPermissions
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Rule
+import org.junit.Test
+
+class PermissionsTest {
+
+    @get:Rule
+    val activityRule = ActivityScenarioRule(SignInActivity::class.java)
+
+    @get:Rule
+    val permissionRule: GrantPermissionRule = GrantPermissionRule.grant(Manifest.permission.CAMERA)
+
+    @Test
+    fun hasPermissionsForCameraReturnsTrueWhenCameraPermissionsAreGranted() {
+        activityRule.scenario.onActivity {
+            assertThat(
+                hasPermissions(it.applicationContext, Manifest.permission.CAMERA),
+                `is`(equalTo(true))
+            )
+        }
+    }
+
+    @Test
+    fun hasPermissionsForBatteryStatsReturnsFalseWhenBatteryStatsPermissionsAreNotGranted() {

```

Comment:
I'm not exactly sure how it works on the CI but I thought that every permission was not granted by default. From the tests I made it seems to be the case, but maybe I'm wrong. Do you happen to know how we could ensure that a permission is not granted?

Reply:
Actually I faced the same issue during a previous sprint. I preferred to not test this case in order to avoid any non-deterministic behavior. Since the test already passes, feel free to keep it.